From 08bc9c8f8bcdcdfa8c9cdba30930b1ba0117d458 Mon Sep 17 00:00:00 2001
From: Kevin Zhang <akz002@freescale.com>
Date: Tue, 22 Jan 2008 15:29:40 -0600
Subject: [PATCH] ENGR00053775 fix suspend to ram wakeup failure

Description:
This patch is to fix the problem of -- System failed to wakeup when suspend to ram with "echo mem > /sys/power/state". This is caused by the spi suspend routine is called too early. The fix is to call the mxc spi suspend routine in a later stage.
---
 drivers/spi/mxc_spi.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/mxc_spi.c b/drivers/spi/mxc_spi.c
index 8f78d77..cccafa4 100644
--- a/drivers/spi/mxc_spi.c
+++ b/drivers/spi/mxc_spi.c
@@ -876,8 +876,8 @@ static struct platform_driver mxc_spi_driver = {
 		   },
 	.probe = mxc_spi_probe,
 	.remove = mxc_spi_remove,
-	.suspend = mxc_spi_suspend,
-	.resume = mxc_spi_resume,
+	.suspend_late = mxc_spi_suspend,
+	.resume_early = mxc_spi_resume,
 };
 
 /*!
-- 
1.5.4.4

